### PR TITLE
Feat: Awmps UI initial

### PR DESCRIPTION
**Pull request by t0risutan**
Singe Page Application changes:

- Added "spa" directory to convert everything into 1 single page application
- Still have old directories and unnecessary code lines which will be deleted
- Some of the new spa implementations are still in progress due to some broken event listeners and missing css

Ps: Mainly pay attention to the "spa" directory

### CODE REVIEW

File: ui/spa/global.html [hunk 1]
```diff
@@ -0,0 +1,27 @@
+<!DOCTYPE html>
+<html lang="en">
+<head>
+    <meta charset="UTF-8">
+    <meta http-equiv="X-UA-Compatible" content="IE=edge">
```
Comment (martinst06) [inline ui/spa/global.html:5]: remove this
(pshhhhh)

File: ui/spa/global.html [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <title>Single Page App</title>
```
Comment (martinst06) [inline ui/spa/global.html]: change title to appropriate page title

File: ui/spa/global.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+</head>
+<body>
+    <nav class="nav">
+        <nav id="sideNav" class="side-nav">
+            <a href="/" class="nav__link" data-link>Home</a>
+            <a href="/watering" class="nav__link" data-link>Watering&nbsp;Schedule</a>
+            <a href="/notifications" class="nav__link" data-link>Notifications</a>
+            <a href="/users" class="nav__link" data-link>Users</a>
+            <a href="/account" class="nav__link" data-link>Account</a>
+            <a href="/login" class="nav__link" data-link>Log-in</a>
```
Comment (martinst06) [inline ui/spa/global.html]: Change to Login instead of Log-in

**Review by martinst06 — CHANGES_REQUESTED** [review]
Please review my comments and fix

File: ui/spa/global.html (hunk 2 above)
Comment (t0risutan) [reply ui/spa/global.html]: ok

**Review by t0risutan — COMMENTED** [review]
fixed fixed fixed and fixed

**Review by martinst06 — APPROVED** [review]

File: ui/.vscode/settings.json
Comment (skholkhojaev) [inline ui/.vscode/settings.json:1]: - remove the vscode and add it to gitignore 
- add the necessary information such as ports in a .env file

**Review by skholkhojaev — CHANGES_REQUESTED** [review]
like said in the comment the vscode folder needs to be changed

**Review by skholkhojaev — APPROVED** [review]

**Pull request merged**